src/easylink/implementation.py

```diff
@@ -24,28 +47,39 @@ def __init__(
         output_slots: Iterable["OutputSlot"] = (),
     ):
         self.name = implementation_config.name
-        """The name of the implementation."""
+        """The name of this Implementation."""
         self.input_slots = {slot.name: slot for slot in input_slots}
-        """A mapping of input slot names to InputSlot instances."""
+        """A mapping of :class:`InputSlots<easylink.graph_components.InputSlot>`
+        names to their instances."""
         self.output_slots = {slot.name: slot for slot in output_slots}
-        """A mapping of output slot names to OutputSlot instances."""
+        """A mapping of :class:`OutputSlots<easylink.graph_components.OutputSlot>` 
+        names to their instances."""
         self._metadata = self._load_metadata()
         self.environment_variables = self._get_env_vars(implementation_config)
         """A mapping of environment variables to set."""
         self.metadata_steps = self._metadata["steps"]
-        """The specific step details that this implementation is associated with."""
+        """The names of the specific :class:`Steps<easylink.step.Step>` for which
+        this Implementation is reesponsible."""
         self.schema_steps = schema_steps
-        """The high-level pipeline schema steps that this implementation is associated with."""
+        """The requested :class:`~easylink.pipeline_schema.PipelineSchema`
+        :class:`~easylink.step.Step` names for which this Implementation is 
+        expected to be responsible."""
```

Comment:
```suggestion
        requested to be responsible in the pipeline."""
```

Reply:
Also, why is this stored as an attribute rather than validated and immediately thrown away?